### ISSUE TITLE
Question: can we take normal conversion losses into account?

### ISSUE THREAD

**Issue by ChaelKruip** [quintel/etmoses#799]
This issue is just to clear up some confusion. 

## Capacities

In ETMoses, we have `capacity` and `carrier_capacity`. It is my understanding that these correspond to the ETE equivalents: `output_capacity` and `input_capacity`. The possible conversion loss of a technology that turns one carrier into another is given by
```
loss = 1 - efficiency
```
where the `efficiency` is defined as
```
efficiency = capacity/carrier_capacity
```

## performance_coefficient

As far as I understand this, the `performance_coefficient` in ETMoses equals the `coefficient_of_performance` in ETE which is 1.0 for all technologies except heat pumps. 

In ETMoses it is used to translate between `capacity` and `carrier_capacity` as in
```
carrier_capacity = capacity / performance_coefficient
```

I think this works correctly for heat pumps but results in `capacity` being (*wrongfully) equal to `carrier_capacity` in all other technologies.

## What to do?

@RobTerwel has researched which numbers would be correct as  `capacity` and `carrier_capacity`, taking into account conversion losses. Two notable exceptions are P2P (batteries) and EV which actually need to lose energy twice (both when *charging* and when *discharging*).

@antw what do you think would be a good (temporary) way to initialise `capacity` and `carrier_capacity` in ETMoses?

Some options that come to mind:
* Add a column to the database with `carrier_capacity` (or `efficiency`) and initialise from there
* Extend the ETE communication functions to import both 'input' and 'output' capacities

**Comment by antw** [comment]
I'd prefer to do it right, and import the input and output capacities directly from ETEngine. I'm not convinced that is sufficient to model loss, however.

In Moses, a capacity is a bottleneck which limits the (maximum) flow of energy, rather than a *conversion* which may have loss. For example, a hypothetical P2P battery with an input capacity of 2kW and output capacity of 1kW charges at full capacity for one hour. 2kWh is stored. It will take *two* hours to fully discharge (2kWh stored / 1kW output capacity = 2 hours). It does not mean that 1kWh is discharged and 1kWh is thrown away as loss.

My feeling is that we want to support input and output efficiency *in addition* to input and output *capacity*.

The `carrier_capacity` attribute is an ugly solution for the lack of an `ambient_heat` input for the heat-pumps; the only place I can guarantee it works is in the [Buffer](https://github.com/quintel/etmoses/blob/fab747abfaee35c3cef2e526253270e9c3397114/app/models/network/technologies/buffer.rb) class which implements the heat-pump behaviour. I would like to be rid of it in favour of a proper capacity/efficiency system.

**Comment by ChaelKruip** [comment]
> In Moses, a capacity is a bottleneck which limits the (maximum) flow of energy, rather than a conversion which may have loss.

Yes. But (!) doesn't it behave as loss for heat producing technologies for instance? If a `households_space_heater_network_gas` produces heat at its maximum `output_capacity` of 22 kW, doesn't it consume gas at 27.5 kW (22 kW/ 0.8)? The discrepancy of 5.2 kW can be interpreted as loss right?

> My feeling is that we want to support input and output efficiency in addition to input and output capacity.

As a longish-term solution I fully agree. But currently, the numbers are wrong and I would like to be able to come up with a temporary solution to allow current users to get somewhat physical results :grimacing:
